dali/base/dadfs.cpp

```diff
@@ -5225,12 +5226,38 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
         init(_parent,tree,_name,user,transaction);
     }
 
+    CDistributedSuperFile(CDistributedFileDirectory *_parent, ISuperFileDescriptor *_desc, IUserDescriptor* user)
+    {
+        parent = _parent;
+        setUserDescriptor(udesc, user);
+        interleaved=0;
+        root.set(_desc->getProperties());
```

Comment:
getProperties() *should* always be returning "Attr" tree...
See other comment in dafdesc.cpp near call to deserializeSuperFileDescriptorTree()

So it's working by accident here at moment, you can't use getProperties() to get XML you need for this constructor..
And you can't use an existing ISuperFileDesciptor as a container for the raw XML, it wasn't intended for that.

From our IM conversation a while back:
```
Attila: How can I create a SuperFile from a SuperFileDescriptor? I  Can't find any constructor.

Jake: there isn't one.
existing createNew() is based on a IFileDescriptor, you could query the IFileDescriptor to see if it's a super IFileDescriptor::querySuperFileDescriptor()
but there is more work to do because ISuperFileDescriptor doesn't supprted all the meta data being imported.
i.e. you would 1st need to create a ISuperFileDescriptor from the import XML that fully represents it, then do createNew()
That would be most 'complete' way of doing it, but means extends (and potentially introducing problems) into the the implementation of ISuperFileDescriptor/CSuperFileDescriptor
You could look at it though and see how much work it would be / how disruptive it would be.
But what I suggest is that you skip creation of a 'full' ISuperFileDescriptor
and introduce a createNewSuper(IPropertyTree *super)
that creates the IDistributeSuperFile you really want directly.

Attila : with CDistributedSuperFile(CDistributedFileDirectory *_parent, IPropertyTree _root,const CDfsLogicalFileName &_name,IUserDescriptor user) constructor?

Jake: yes I think so, but don't think it needs name or user..
can't remember, might be supplied at attach time, let me look Sep 25, 2015 12:42
as until attached, it's  standalone, needs no authority(user)
yes attach takes user name, which makes sense
ah you were talking about an existing constructor, not a new one...
let me look at it, see what it does
ok, I see (to some extent) it needs user, because it looks up subfiles...
this is a bit different, it is creating a CDistributedSuperFile based on an existing published file.
may be very similar though
the only difference may be, that the name shouldn't be supplied.
does that make sense?
It should probably not load the subfiles upfront either.. loading means locking..
which would mean an unattached super, locks published subfiles, which would break the detach()'s semantics of a file.
So a new constructor:
CDistributedSuperFile(CDistributedFileDirectory *_parent, IPropertyTree _root);
which does similar init, except delays loadSubFiles to point where it's attach()'d
I'm sure I've glossed over a few details, but I think that's broadly correct
```


So as we discussed before, you should pass the IPropertyTree in directly to your new CDistributedSuperFile constructor.

Also mentioned above, the new constructor should be be very similar to the existing constructor
```
    CDistributedSuperFile(CDistributedFileDirectory *_parent, IPropertyTree *_root,const CDfsLogicalFileName &_name,IUserDescriptor* user) 
```
except:
1) You are creating an unattached file (not an object based on a published file), so no need for 'name' or 'user', that comes during attach()
2) You do not want the init to loadSubFiles, because that will lookup and lock them and you are creating an unattached file (no Dali connection at this point). It should delay that action until attach().

Your new constructor code should look something like:
```
    void commonInit(IPropertyTree *_root)
    {   
        root.set(_root);
        const char *val = root->queryProp("@interleaved");
        if (val&&isdigit(*val))
            interleaved = atoi(val);
        else
            interleaved = strToBool(val)?1:0;
    }

    CDistributedSuperFile(CDistributedFileDirectory *_parent, IPropertyTree *_root) // constructor for unattached superfile
    {
      commonInit(root);
    }
```

the existing init() can probably be changed to call commonInit() as doing common work.

The rest of the initialization should happen during attach() , i.e. the loadSubFiles and the linking.